conda/utils.py

```diff
@@ -266,8 +266,26 @@ def wrap_subprocess_call(
                         "requirements)"
                     )
                 fh.write(f"{silencer}{quote_for_shell(*arguments)}\n")
-            fh.write(f"{silencer}IF %ERRORLEVEL% NEQ 0 EXIT /b %ERRORLEVEL%\n")
+            # Capture the user's command exit code before deactivation.
+            fh.write(f"{silencer}SET _CONDA_EXE_RC=%ERRORLEVEL%\n")
+
+            fh.write(f"{silencer}IF DEFINED CONDA_PREFIX (\n")
+            fh.write(
+                f'{silencer}  IF EXIST "%CONDA_PREFIX%\\etc\\conda\\deactivate.d" (\n'
+            )
+            fh.write(
+                f'{silencer}    FOR /F "delims=" %%S IN (\'dir /b /a:-d /o:-n "%CONDA_PREFIX%\\etc\\conda\\deactivate.d\\*.bat"\') DO (\n'
+            )
+            fh.write(
+                f'{silencer}      IF EXIST "%CONDA_PREFIX%\\etc\\conda\\deactivate.d\\%%S" CALL "%CONDA_PREFIX%\\etc\\conda\\deactivate.d\\%%S"\n'
```

Comment:
If the script was returned by the glob, it must exist, so no need to check again, right? I would understand you want to make sure it's a file and not a directory, but that's not what's being checked here so... let's CALL it right away?